test/jdk/java/awt/print/PrinterJob/PrintTextTest.java

```diff
@@ -21,193 +21,212 @@
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.GraphicsEnvironment;
+import java.awt.RenderingHints;
+import java.awt.font.FontRenderContext;
+import java.awt.font.GlyphVector;
+import java.awt.font.TextLayout;
+import java.awt.geom.AffineTransform;
+import java.awt.geom.Point2D;
+import java.awt.print.Book;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import java.text.AttributedCharacterIterator;
+import java.text.AttributedString;
+import java.util.HashMap;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JTabbedPane;
+
+public class PrintTextTest {
+
+    static final String INSTRUCTIONS = """
+        This tests that printed text renders similarly to on-screen under a variety
+        of APIs and graphics and font transforms.
+        1. Print to your preferred printer.
+        2. Collect the output.
+        3. Refer to the onscreen buttons to cycle through the on-screen content.
+        4. For each page, confirm that the printed content corresponds to the
+           on-screen rendering for that *same* page. Some cases may look odd but
+           its intentional. Verify it looks the same on screen and on the printer.
+        Note that text does not scale linearly from screen to printer so some
+        differences are normal and not a bug.
+        The easiest way to spot real problems is to check that any underlines are
+        the same length as the underlined text and that any rotations are the same
+        in each case.
+        Note that each on-screen page is printed in both portrait and landscape mode.
+        So for example, Page 1/Portrait, and Page 1/Landscape when rotated to view
+        properly, should both match Page 1 on screen.
+        """;
+
+    public static void main(String[] args) throws Exception {
 
         PrinterJob pjob = PrinterJob.getPrinterJob();
         PageFormat portrait = pjob.defaultPage();
         portrait.setOrientation(PageFormat.PORTRAIT);
-        preferredSize = (int)portrait.getImageableWidth();
+        int preferredSize = (int) portrait.getImageableWidth();
 
         PageFormat landscape = pjob.defaultPage();
         landscape.setOrientation(PageFormat.LANDSCAPE);
 
         Book book = new Book();
 
-        JTabbedPane p = new JTabbedPane();
+        JTabbedPane pane = new JTabbedPane();
 
         int page = 1;
         Font font = new Font("Dialog", Font.PLAIN, 18);
-        String name = "Page " + new Integer(page++);
-        PrintTextTest ptt = new PrintTextTest(name, font, null, false);
-        p.add(name, ptt);
-        book.append(ptt, portrait);
-        book.append(ptt, landscape);
+        String name = "Page " + page++;
+        PrintText pt = new PrintText(name, font, null, false, preferredSize);
+        pane.add(name, pt);
```

Comment:
```suggestion
        pane.addTab(name, pt);
```
I think [`JTabbedPane.addTab`](https://docs.oracle.com/en/java/javase/21/docs/api/java.desktop/javax/swing/JTabbedPane.html#addTab(java.lang.String,java.awt.Component)) should be used to add components rather than inherited `add`.